### PR TITLE
added .gitignore, github action & github templates

### PR DESCRIPTION
**Pull request by monotek**
Signed-off-by: André Bauer <monotek23@gmail.com>

* added gitignore file
* added github issue template
* added github pr template
* added github action for pr ci 
  * lints all bash scripts
  * uses helm template & kubeval to validate created k8s schemas (https://github.com/instrumenta/kubeval/)
  * lints chart (https://github.com/helm/chart-testing-action)
  * installs chart in a kind cluster (https://github.com/helm/kind-action & https://github.com/helm/chart-testing-action)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<!--
+Thank you for contributing to kiwigrid/charts. Before you submit this PR we'd like to
```
Comment (spencergilbert) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion
Thank you for contributing to thanos-community/helm-charts. Before you submit this PR we'd like to
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Following our best practices right from the start will accelerate the review process and
+help get your PR merged quicker.
+
+When updates to your PR are requested, please add new commits and do not squash the
+history. This will make it easier to identify new changes. The PR will be squashed
+anyways when it is merged. Thanks.
+
+For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.
+
+Please make sure you test your changes before you push them. Once pushed, a CircleCI
```
Comment (spencergilbert) [inline .github/PULL_REQUEST_TEMPLATE.md:22]:
```suggestion
Please make sure you test your changes before you push them. Once pushed, a GitHub Actions pipeline
```
?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+#### Special notes for your reviewer:
+
+
+#### Checklist
+[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
+- [ ] [DCO](https://developercertificate.org) signed
+- [ ] Chart Version bumped (if the pr is an update to an existing chart)
+- [ ] Variables are documented in the README.md
+- [ ] Title of the PR starts with chart name (e.g. `[fluentd-elasticsearch]`)
```
Comment (spencergilbert) [inline .github/PULL_REQUEST_TEMPLATE.md:44]: Possibly unneeded if we just have a single chart

File: .github/ct.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# consider helm install to be failed after 10 minutes
+helm-extra-args: --timeout 600
+check-version-increment: true
+debug: true
+chart-dirs: 
+  - charts
+chart-repos:
+  - stable=https://kubernetes-charts.storage.googleapis.com/
+  - kiwigrid=https://kiwigrid.github.io
```
Comment (spencergilbert) [inline .github/ct.yaml:9]: don't have a thanos repo yet but...

File: .github/kubeval.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+curl --silent --show-error --fail --location --output get_helm.sh https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get
+chmod 700 get_helm.sh
+./get_helm.sh --version "${HELM_VERSION}"
+
+# install kubeval
+curl --silent --show-error --fail --location --output /tmp/kubeval.tar.gz https://github.com/instrumenta/kubeval/releases/download/"${KUBEVAL_VERSION}"/kubeval-linux-amd64.tar.gz
+sudo tar -C /usr/local/bin -xf /tmp/kubeval.tar.gz kubeval
+
+# add helm repos to resolve dependencies
+helm repo add kiwigrid https://kiwigrid.github.io
```
Comment (spencergilbert) [inline .github/kubeval.sh:24]: don't have a thanos repo yet but...

**Comment by spencergilbert** [comment]
@monotek I assume you pulled this from an existing setup using github actions? I haven't used it before but it looks thorough, and i'd love to dive in and learn a bit more there

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (monotek) [reply .github/PULL_REQUEST_TEMPLATE.md:44]: True

File: .github/ct.yaml (hunk 4 above)
Comment (monotek) [reply .github/ct.yaml:9]: These repos are only needed for dependencies.
Lets see if we need them at all. The kiwigrid thanso chart uses our any-resource chart to add some resources, therefore its there. I think i'll remove for now.

**Comment by monotek** [comment]
> @monotek I assume you pulled this from an existing setup using github actions? I haven't used it before but it looks thorough, and i'd love to dive in and learn a bit more there

yes, mostly copied from https://github.com/kiwigrid/helm-charts

**Comment by spencergilbert** [comment]
> > @monotek I assume you pulled this from an existing setup using github actions? I haven't used it before but it looks thorough, and i'd love to dive in and learn a bit more there
> 
> yes, mostly copied from https://github.com/kiwigrid/helm-charts

Thanks - I haven't looked through those before, I'll take a look and also at that any-resource chart. Very interesting.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (bwplotka) [reply .github/PULL_REQUEST_TEMPLATE.md:44]: We need DCO as well though... plus we might have more than one, depending on the outcome of https://docs.google.com/document/d/18GXxwOm9c2fDK20LV1Sekte0xvK4qXpIGwBcuw_ZYSA/edit?ts=5e9b2c6e#

**Comment by bwplotka** [comment]
It does!

**Pull request merged**